src/WebJobs.Script/Description/Workers/WorkerFunctionDescriptorProvider.cs

```diff
@@ -52,10 +54,20 @@ protected override IFunctionInvoker CreateFunctionInvoker(string scriptFilePath,
         protected override async Task<Collection<ParameterDescriptor>> GetFunctionParametersAsync(IFunctionInvoker functionInvoker, FunctionMetadata functionMetadata,
             BindingMetadata triggerMetadata, Collection<CustomAttributeBuilder> methodAttributes, Collection<FunctionBinding> inputBindings, Collection<FunctionBinding> outputBindings)
         {
-            var parameters = await base.GetFunctionParametersAsync(functionInvoker, functionMetadata, triggerMetadata, methodAttributes, inputBindings, outputBindings);
-
             var bindings = inputBindings.Union(outputBindings);
 
+            // If an input or output binding uses expressions, we do not want to use ParameterBindingData for the trigger bindings
+            if (triggerMetadata.SupportsDeferredBinding())
+            {
+                bool skipDeferBinding = BindingAttributeContainsExpression(bindings);
+                if (skipDeferBinding)
+                {
+                    triggerMetadata.Properties.Add(BindingMetadataExtensions.SkipDeferredBindingKey, true);
```

Comment:
So we're already mutating the BindingMetadata here by adding this new property. I wonder if it'd be any worse to just remove the "supports" property in this case, that way we don't need the complexity of 2 different flags?

Reply:
Yeah sounds to me

Reply:
What you have is OK. The two flags are a little confusing, and I was trying to see if we could improve, but I don't have a solid proposal.